File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-20/automation/2026-07-20-control-evidence-evidence-index-validation-tool.ts
/** Synthetic AC-27 validation helper for p03-full-007827; no network or file access. */
const organization = "Cobalt Harbor Systems";
const workingPaperContext = { stage: "follow-up", evidenceCard: "EV-184-04", reviewLane: "PRIV-04", verificationPass: 7, workingPaperSequence: 7827, spineIds: ["org.cobalt-harbor", "evidence.set-184", "rel.ac27.evidence-184", "date.2026-07-20", "term.no-kio-claim"], rationale: "Seven-day closure follow-up confirmed the residual-risk status and checksum reconciliation.", observation: "Evidence indexing links bounded evidence references to the AC-27 review and retention disposition.", canonicalValues: ["Evidence set 184 contains 24 synthetic items: 12 access-review attestations, 6 approval records, 4 checksum manifests, and 2 exception records."] };
type EvidenceRecord = { control_id: string; case_id: string; risk_id: string; evidence_set: number; reviewed: number; timely_attestations: number; residual_risk: string };
const expected: EvidenceRecord = { control_id: "AC-27", case_id: "CASE-260713-184", risk_id: "RISK-031", evidence_set: 184, reviewed: 48, timely_attestations: 47, residual_risk: "Medium" };
export function validate(record: EvidenceRecord): string[] { return Object.keys(expected).filter((key) => record[key as keyof EvidenceRecord] !== expected[key as keyof EvidenceRecord]); }
void organization; void workingPaperContext;

File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-20/automation/2026-07-20-control-evidence-evidence-index-validation-tool.rs
// Synthetic AC-27 validation helper for p03-full-007027; no network or file access.
// stage=follow-up; card=EV-184-20; lane=PRIV-20; pass=10; working_paper=7027; spine=org.cobalt-harbor, evidence.ev-184-19, rel.ac27.evidence-184, date.2026-07-20, term.no-kio-claim
// Seven-day closure follow-up confirmed the residual-risk status and checksum reconciliation. Evidence indexing links bounded evidence references to the AC-27 review and retention disposition.
const ORGANIZATION: &str = "Cobalt Harbor Systems";
pub fn valid(control_id: &str, case_id: &str, risk_id: &str, evidence_set: u32, reviewed: u32, timely: u32, residual: &str) -> bool {
    let _organization = ORGANIZATION;
    control_id == "AC-27" && case_id == "CASE-260713-184" && risk_id == "RISK-031" && evidence_set == 184 && reviewed == 48 && timely == 47 && residual == "Medium"
}
